### PR TITLE
WINDUPRULE-613: fix moved components rule

### PR DESCRIPTION
**Pull request by mmelko**
https://issues.redhat.com/browse/WINDUPRULE-613

### CODE REVIEW

**Comment by jonathanvila** [comment]
LGTM :+1:

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.groovy [hunk 1]
```diff
@@ -125,28 +125,30 @@ ruleSet("xml-moved-components-groovy")
...
                         .filter { component ->
                             (movedComponents[component][0] != "" ? xmlCondition.apply(movedComponents[component][0], "http://camel.apache.org/schema/spring").evaluate(event, context) : false) ||
                                     (movedComponents[component][0] != "" ? xmlConditionWithXpath.apply(movedComponents[component][0],
                                             "http://camel.apache.org/schema/blueprint").evaluate(event, context) : false) ||
                                     (movedComponents[component][1] != "" ? javaConditionMatches.apply(movedComponents[component][1]).evaluate(event, context) : false)
                         }
-                        .each { component -> usedComponents.add(component.replaceAll("\\s","")) }
+                        .each { component -> usedComponents.add(component.replaceAll("\\s", "")) }
 
                 String components = usedComponents.stream()
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.groovy:137]:
```suggestion
                if (usedComponents.size > 0) {
                   String components = usedComponents.stream()
```

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.groovy [hunk 2]
```diff
@@ -125,28 +125,30 @@ ruleSet("xml-moved-components-groovy")
...
-                        .withText(" $components components were moved out of `camel-core` to separate artifacts. Maven users of Apache Camel can keep using the dependency `camel-core`" +
-                                "which has transitive dependencies on all of its modules, except for `camel-main`" +
-                                "and therefore no migration is needed. However, users who want to trim the size of the classes on the classpath," +
-                                "can use fine grained Maven dependencies on only the modules needed.")
-                        .withIssueCategory(optionalIssueCategory)
-                        .with(modularizationLink)
-                        .withEffort(1))
-                        .perform(event, context, payload)
-
+                if (components.length() > 0) {
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.groovy:140]:
```suggestion

```

**Comment by jonathanvila** [comment]
The expression in the method `xmlCondition` should be changed as it is only searching for the component exactly in the 1st child of `route`:
`            .matchesXpath("//*/c:route//*[starts-with(@uri, '$component:')]")`

**Comment by mmelko** [comment]
> The expression in the method `xmlCondition` should be changed as it is only searching for the component exactly in the 1st child of `route`:
> ` .matchesXpath("//*/c:route//*[starts-with(@uri, '$component:')]")`

actually not really,  you can have also other parents  for `<to>` for example:
```xml
<route>
            <from uri="direct:start"/>
            <multicast stopOnException="true">
                <to uri="direct:foo"/>
                <to uri="direct:bar"/>
                <to uri="direct:baz"/>
            </multicast>
            <to uri="mock:result"/>
        </route>
```

**Comment by jonathanvila** [comment]
> > The expression in the method `xmlCondition` should be changed as it is only searching for the component exactly in the 1st child of `route`:
> ` .matchesXpath("//*/c:route//*[starts-with(@uri, '$component:')]")`

actually not really,  you can have also other parents  for `<to>` for example:
```xml
<route>
            <from uri="direct:start"/>
            <multicast stopOnException="true">
                <to uri="direct:foo"/>
                <to uri="direct:bar"/>
                <to uri="direct:baz"/>
            </multicast>
            <to uri="mock:result"/>
        </route>
``` 
What I mean is that it should be changed from c:route/*[starts-   to c:route//*[starts-    in order to search in any level. Now with your code it is only searching on the first child level

**Comment by jonathanvila** [comment]
LGTM :+1:

**Comment by jonathanvila** [comment]
retest

**Comment by jonathanvila** [comment]
Retest

**Comment by jonathanvila** [comment]
Retest

**Pull request merged**